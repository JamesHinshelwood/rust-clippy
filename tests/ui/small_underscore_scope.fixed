// run-rustfix

#![warn(clippy::small_underscore_scope)]

struct NewType(u8);

#[allow(dead_code)]
struct Foo {
    x: u8,
    y: u8,
}

fn main() {
    let n = NewType(24);
    let _ = n;
    let _ = n;

    let t = (4, 5);
    let _ = t;
    let _ = t;

    let t2 = (6, (7, 8));
    let (_x, _) = t2;

    let f = Foo { x: 3, y: 7 };
    let _ = f;
}
