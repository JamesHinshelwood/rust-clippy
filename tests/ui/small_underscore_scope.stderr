error: this wildcard binding could have a wider scope
  --> $DIR/small_underscore_scope.rs:15:9
   |
LL |     let NewType(_) = n;
   |         ^^^^^^^^^^ help: try: `_`
   |
   = note: `-D clippy::small-underscore-scope` implied by `-D warnings`

error: this wildcard binding could have a wider scope
  --> $DIR/small_underscore_scope.rs:16:9
   |
LL |     let NewType(..) = n;
   |         ^^^^^^^^^^^ help: try: `_`

error: this wildcard binding could have a wider scope
  --> $DIR/small_underscore_scope.rs:19:9
   |
LL |     let (_, _) = t;
   |         ^^^^^^ help: try: `_`

error: this wildcard binding could have a wider scope
  --> $DIR/small_underscore_scope.rs:20:9
   |
LL |     let (..) = t;
   |         ^^^^ help: try: `_`

error: this wildcard binding could have a wider scope
  --> $DIR/small_underscore_scope.rs:23:14
   |
LL |     let (_x, (_, _)) = t2;
   |              ^^^^^^ help: try: `_`

error: this wildcard binding could have a wider scope
  --> $DIR/small_underscore_scope.rs:26:9
   |
LL |     let Foo { .. } = f;
   |         ^^^^^^^^^^ help: try: `_`

error: aborting due to 6 previous errors

